CIScripts/Build/BuildFunctions.ps1

```diff
@@ -270,12 +268,11 @@ function Run-Test {
     Param ([Parameter(Mandatory = $true)] [String] $TestExecutable)
     Write-Host "===> Agent tests: running $TestExecutable..."
     $Res = Invoke-Command -ScriptBlock {
-        $ErrorActionPreference = "SilentlyContinue"
-        $TestOutput = Invoke-NativeCommand -AllowNonZero $true -ScriptBlock {
+        $NativeCommandReturn = Invoke-NativeCommand -AllowNonZero $true -ScriptBlock {
             Invoke-Expression $TestExecutable
         }
-        $ExitCode = $TestOutput[-1]
-        $TestOutput.ForEach({ Write-Host $_ })
+        $ExitCode = $NativeCommandReturn[-1]
+        $TestOutput = $NativeCommandReturn[0..($NativeCommandReturn.Length-2)]
```

Comment:
This isn't really clear, but it's not really part of this PR, so fix it if you want.

Maybe we should add:
1. `CaptureOutput` flag to `Invoke-NativeCommand`
2. Return a structure from `Invoke-NativeCommand` with two fields: `ExitCode` and `TestOutput`
3. If `CaptureOutput` is `$false`, then `TestOutput` would be `$null`

Reply:
> If CaptureOutput is $false, then TestOutput would be $null

And it should be live-streamed through `Write-Host`

Reply:
I tried to do this, it's actually pretty hard because PS never does what you expect it to do, so I won't do this.